### PR TITLE
NIFI-4535 Only update Page Title to root flow name when user has perm…

### PR DESCRIPTION
**Pull request by patricker**
…ission.

Updated to handle scenario where user does not have read access.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas.js [hunk 1]
```diff
@@ -167,12 +167,14 @@
...
-                rootBreadcrumb = rootBreadcrumb.parentBreadcrumb
-            }
+            if (breadcrumb.permissions.canRead) {
+                var rootBreadcrumb = breadcrumb;
+                while(rootBreadcrumb.parentBreadcrumb != null) {
+                    rootBreadcrumb = rootBreadcrumb.parentBreadcrumb
+                }
 
-            document.title = rootBreadcrumb.breadcrumb.name;
+                document.title = rootBreadcrumb.breadcrumb.name;
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas.js]: @patricker Thanks for the PR!

I haven't run this yet, but I think there may still be an issue. We need to ensure we have permissions to the `rootBreadcrumb` here. If the user navigated to a sub-Process Group but did not have permissions to the root Process Group I think we'd still have the error.

Additionally, if the user navigated to a sub-Process Group which they didn't have access to but they did have access to the root Process Group we would unnecessarily avoid setting the document title.

I don't believe there was an issue with identifying the `rootBreadcrumb`. The issue was attempting to extract the breadcrumb's name through the `breadcrumb` field which is `null`ed out when a user doesn't have permissions to the corresponding Process Group.

Comment (patricker) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas.js]: I really appreciate your deep understanding of these more corner case security issues!
I'm thinking on it, will hope to have it updated Friday.

**Comment by patricker** [comment]
@mcgilman I played around with permissions, and I think this one covers all the cases you mentioned. It will loop up the breadcrumb tree, which you can do even if you have no read access, and then checks at the root level if read access exists.

**Comment by mcgilman** [comment]
Thanks @patricker! This has been merged to master.

**Pull request closed without merging**